mono/metadata/profiler-private.h

```diff
@@ -57,6 +57,12 @@ void mono_profiler_module_loaded (MonoImage *image, int result);
 void mono_profiler_class_event  (MonoClass *klass, int code);
 void mono_profiler_class_loaded (MonoClass *klass, int result);
 
+void mono_profiler_static_field (MonoClassField *field);
+void mono_profiler_static_fields_allocated (MonoClass *klass);
+
+void mono_profiler_special_static_area_allocated (MonoObject *owner, gboolean thread_local, gsize owner_id, int index, void **addr, gsize size);
```

Comment:
This "gboolean thread_local" is a horrible design. It assumes that there are 2 options: thread local and the opposite of that which is, hmmm, global?

Since this is to signal between thread local/context local, let's use an enum so it's clear what each value means.